src/main/java/org/openrewrite/staticanalysis/ReplaceDuplicateStringLiterals.java

```diff
@@ -181,7 +187,16 @@ private String transformToVariableName(String valueOfLiteral) {
                         prevIsLower = Character.isLowerCase(c);
                     }
                 }
-                return VariableNameUtils.normalizeName(newName.toString());
+                String newNameString = newName.toString();
+                while(newNameString.length() > maxVariableLength){
+                    int indexOf = newNameString.lastIndexOf("_");
+                    if(indexOf > -1) {
+                        newNameString = newNameString.substring(0, indexOf);
+                    } else{
```

Comment:
I see you've adapted the Sam-style of formatting PRs ;)
```suggestion
                while (newNameString.length() > maxVariableLength){
                    int indexOf = newNameString.lastIndexOf("_");
                    if (indexOf > -1) {
                        newNameString = newNameString.substring(0, indexOf);
                    } else {
```